Realm/RLMQueryUtil.mm

```diff
@@ -768,8 +770,8 @@ void validate_and_extract_between_range(id value, RLMProperty *prop, id *from, i
         // For arrays this effectively checks if there are any objects in the
         // array, while for links it's just always constant true or false
         // (for != and = respectively).
-        if (column.property().collection) {
-            add_bool_constraint(RLMPropertyTypeObject, operatorType, column.resolve<Link>(), null());
+        if (!column.link_map().only_unary_links()) {
```

Comment:
I don't think this change is correct. `("array.link = %@", unmanagedObject)` should be FalseExpression.

Reply:
This now uses what is in master, is master not correct? Just confused as it says this comment is Outdated.

Reply:
Oh, that's a change from merging in master and I got confused by looking at the changes since last review. I think that's my bug then...